xapian-core/backends/glass/glass_database.cc

```diff
@@ -1049,8 +1050,11 @@ GlassWritableDatabase::GlassWritableDatabase(const string &dir, int flags,
     LOGCALL_CTOR(DB, "GlassWritableDatabase", dir | flags | block_size);
 
     const char *p = getenv("XAPIAN_FLUSH_THRESHOLD");
-    if (p)
-	flush_threshold = atoi(p);
+    if (p) {
+	if (!parse_signed(p, flush_threshold)) {
```

Comment:
Again, I think an empty environment variable should continue to work the same as not setting it.

And a negative threshold doesn't make much sense, so I think this should use `parse_unsigned`.

Reply:
fixed in a fixup commit.